infra/gcp/terraform/kubernetes-public/k8s-prod-readiness.tf

```
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+ 
+/*
+This file defines:
+- A BigQuery dataset for prod-readiness reporting
+- IAM bindings on that dataset
+*/
+
+locals {
+  prod_readiness_owners = "k8s-infra-prod-readiness@kubernetes.io"
+}
+
+// BigQuery dataset for PRR analysis
+resource "google_bigquery_dataset" "prod_readiness_dataset" {
+  dataset_id  = "k8s_prod_readiness"
+  project     = data.google_project.project.project_id
+  description = "Dataset for prod-readiness approvers to use for reporting purposes"
+  location    = "US"
+
+  // Data is precious, make it difficult to delete by accident
+  delete_contents_on_destroy = false
+}
+
+data "google_iam_policy" "prod_readiness_dataset_iam_policy" {
+  binding {
+    members = [
+      "group:${local.prod_readiness_owners}",
+    ]
+    role = "roles/bigquery.dataOwner"
+  }
+}
+
+resource "google_bigquery_dataset_iam_policy" "prod_readiness_dataset" {
+  dataset_id  = google_bigquery_dataset.prod_readiness_dataset.dataset_id
+  project = google_bigquery_dataset.prod_readiness_dataset.project
```

```suggestion
  project = data.google_project.project.project_id
```
I may be wrong `google_bigquery_dataset` don't have this attribute `project` : https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_dataset#attributes-reference

---

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_dataset#project

I've used this before for `triage_dataset`

---

As a style thing, I'd rather tie it the project containing the dataset vs. the project this module happens to be in charge of.  I think that makes it harder for the two resources to fall out of sync (e.g. change the project setting on one but not the other)